src/XMakeBuildEngine/Evaluation/Expander.cs

```diff
@@ -2609,6 +2609,71 @@ internal enum InvokeType
         }
 #endif
 
+        private class FunctionBuilder<T>
```

Comment:
I might find the codeflow below clearer if this was a struct passed by ref--then the void methods that alter it in place would jump out to me more.